oauth2_http/javatests/com/google/auth/oauth2/MockMetadataServerTransport.java

```diff
@@ -164,7 +164,11 @@ public LowLevelHttpResponse execute() throws IOException {
         // Create the JSON response
         GenericJson refreshContents = new GenericJson();
         refreshContents.setFactory(OAuth2Utils.JSON_FACTORY);
-        refreshContents.put("access_token", accessToken);
+        if (url.contains("?scopes=")) {
+          refreshContents.put("access_token", "fake access token with scope");
```

Comment:
If possible, could the fake access token not be hard coded in the MockMetaDataServerTransport code?

Perhaps we could make the logic something like:
1. MockMetaDataServerTransport takes a map of scopes -> access token pairings/ MockMetaDataServerTransport.addAccessTokenScope(scope, accessToken) will add the the pairing
2. If url.contains("?scopes="), check the map for scopes and returns the configured access token
3. Test passes the pairing of (scope, accesstoken) to MockMetaDataServerTransport